suite-common/metadata-types/src/metadataTypes.ts

```diff
@@ -139,30 +139,62 @@ export abstract class AbstractMetadataProvider {
     }
 }
 
-export interface AccountMetadata {
-    key: string; // legacy xpub format (btc-like coins) or account descriptor (other coins)
-    fileName: string; // file name in dropbox
-    aesKey: string; // asymmetric key for file encryption
+export interface AccountLabels {
     accountLabel?: MetadataItem;
     outputLabels: { [txid: string]: { [index: string]: MetadataItem } };
     addressLabels: { [address: string]: MetadataItem };
 }
 
+export interface WalletLabels {
+    walletLabel?: string;
+}
+
+export type Labels = AccountLabels | WalletLabels;
+
 export type DeviceMetadata =
     | {
           status: 'disabled' | 'cancelled'; // user rejects "Enable labeling" on device
       }
-    | {
+    | ({
           status: 'enabled';
-          key: string; // master key for all values (Device and Account)
-          fileName: string; // file name in dropbox
-          aesKey: string; // asymmetric key for file encryption
-          walletLabel?: string;
-      };
+      } & LabelableEntityKeysByVersion);
+
+type Data = Record<
+    string, // key is filename
+    Labels
+>;
+
+type ClientId = string;
+
+/**
+ * DataType dictates shape of data.
+ * in the future, it could be
+ * 'labels' | 'passwords' | 'contacts'...
+ */
+export type DataType = 'labels';
+
+/**
+ * Representation of provider data stored in reducer
+ * properties 'tokens' and 'type' are needed to recreate corresponding provider instance
+ * others may be used in UI
+ */
+export type MetadataProvider = {
+    type: MetadataProviderType;
+    user: string;
+    tokens: Tokens;
+    isCloud: boolean;
+    // decrypted content of data per provider
+    data: Data;
+    clientId: ClientId;
+};
 
 export interface MetadataState {
     enabled: boolean; // global for all devices
-    provider?: MetadataProvider;
+    providers: MetadataProvider[];
+    // being selected means:
+    // - see data from this provider
+    // - save data to this provider when making changes
+    selectedProvider: { [key in DataType]: ClientId };
```

Comment:
Not a strong opinion. However, it would look better if you use `MetadataProvider["clientId"]` and just type it in `MetadataProvider` as `string` instead of doing `type ClientId = string;`

Reply:
ok why not 5a812239b2262b5c1005319e9d115364e4ef8e85